src/sonic-py-common/sonic_py_common/multi_asic.py

```diff
@@ -480,3 +482,31 @@ def validate_namespace(namespace):
         return True
     else:
         return False
+
+def get_asic_presence_list():
+    """
+    @summary: This function will get the asic presence list. On Supervisor, the list includes only the asics
+              for inserted and detected fabric cards. For non-supervisor cards, e.g. line card, the list should
+              contain all supported asics by the card. The function gets the asic list from CHASSIS_ASIC_TABLE from
+              CHASSIS_STATE_DB. The function assumes that the first N asic ids (asic0 to asic(N-1)) in
+              CHASSIS_ASIC_TABLE belongs to the supervisor, where N is the max number of asics supported by the Chassis 
+    @return:  List of asics present
+    """
+    asics_list = []
+    if is_multi_asic():
+        if not is_supervisor():
+            # This is not supervisor, all asics should be present. Assuming that asics
+            # are not removable entity on Line Cards. Add all asics, 0 - num_asics to the list.
+            asics_list = list(range(0,get_num_asics()))
+        else:
+            # This is supervisor card. Some fabric cards may not be inserted.
+            # Get asic list from CHASSIS_ASIC_TABLE which lists only the asics
+            # present based on Fabric card detection by the platform.
+            db = daemon_base.db_connect(CHASSIS_STATE_DB)
```

Comment:
Do we really need daemon_base to be included and used. We could just use SonicV2Connector itself.

Reply:
@judyjoseph , as discussed, will use swsscommon.DBConnector() directly here instead of importing the function from daemon_base